### PR TITLE
lint: fix lint errors in pkg br.

### PR DESCRIPTION
**Pull request by 3pointer**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/31358

Problem Summary:
run `golangci-lint run --out-format checkstyle  --tests=false` and use the default tidb lint config ` .golangci.yml `

### What is changed and how it works?
fix lint error

### Check List

Tests 

- [ ] Unit test
- [ ] No code

### Release note

```release-note
None
```

### CODE REVIEW

File: br/pkg/lightning/restore/check_info.go [hunk 1]
```diff
@@ -1011,7 +1016,6 @@ outloop:
 				initializedColumns = true
 			}
 		case io.EOF:
-			reachEOF = true
```
Comment (glorv) [inline br/pkg/lightning/restore/check_info.go:1014]: this variable can be removed after this change

File: br/pkg/lightning/restore/restore.go [hunk 2]
```diff
@@ -1912,8 +1912,7 @@ func (rc *Controller) preCheckRequirements(ctx context.Context) error {
 					return errors.Trace(err)
 				}
 				if err := rc.clusterResource(ctx, source); err != nil {
-					rc.taskMgr.CleanupTask(ctx)
-					return errors.Trace(err)
+					return rc.taskMgr.CleanupTask(ctx)
```
Comment (glorv) [inline br/pkg/lightning/restore/restore.go:1915]:
```suggestion
					if err1 := rc.taskMgr.CleanupTask(ctx); err1 != nil {
						 log.Warn(...)
					}
					return err
```

File: br/pkg/lightning/restore/restore.go [hunk 3]
```diff
@@ -1927,7 +1926,10 @@ func (rc *Controller) preCheckRequirements(ctx context.Context) error {
 	}
 	if !rc.checkTemplate.Success() {
 		if !taskExist && rc.taskMgr != nil {
-			rc.taskMgr.CleanupTask(ctx)
+			err := rc.taskMgr.CleanupTask(ctx)
+			if err != nil {
+				return errors.Trace(err)
```
Comment (glorv) [inline br/pkg/lightning/restore/restore.go:1931]:
```suggestion
				log.L().Warn(...)
```

**Review by glorv — COMMENTED** [review]
rest LGTM

File: br/pkg/backup/client.go [hunk 4]
```diff
@@ -951,26 +954,29 @@ backupLoop:
 				zap.Int("retry time", retry))
 			return berrors.ErrFailedToConnect.Wrap(err).GenWithStack("failed to create backup stream to store %d", storeID)
 		}
-		defer bcli.CloseSend()
```
Comment (kennytm) [inline br/pkg/backup/client.go:954]: can this be

```go
defer func() {
    _ = bcli.CloseSend()
}()
```

rather than copying the whole thing everything

**Review by kennytm — APPROVED** [review]
rest lgtm

File: br/pkg/backup/client.go (hunk 4 above)
Comment (3pointer) [reply br/pkg/backup/client.go:954]: this defer is in a for loop. 
I will use another PR to refactor this function. https://github.com/pingcap/tidb/pull/31264

**Review by YuJuncen — APPROVED** [review]

**Pull request merged**